### PR TITLE
Dash financial report

### PR DESCRIPTION
**Pull request by celinehuang**
Issue for app: https://github.com/plotly/dash-sample-apps/issues/10

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL:  https://dash-playground.plotly.host/dash-financial-report/
- Current gallery app URL: https://dash-gallery.plotly.host/dash-financial-report/

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [x] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

**Comment by celinehuang** [comment]
# Note 
The header is different from the mock up but it has been approved by Sidi.

dash-financial-report-thumbnail

File: apps/dash-financial-report/app.py [hunk 1]
```diff
@@ -88,8 +66,6 @@ def display_page(pathname):
 
 
 # detail the way that external_css and external_js work and link to alternative method locally hosted
```
Comment (shammamah) [inline apps/dash-financial-report/app.py]: Please get rid of this comment too!

File: apps/dash-financial-report/app.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import pandas as pd
+
+# import pathlib
+
+app = dash.Dash(
+    __name__, meta_tags=[{"name": "viewport", "content": "width=device-width"}]
+)
+server = app.server
+
+# Describe the layout/ UI, of the app
```
Comment (shammamah) [inline apps/dash-financial-report/app.py]:
```suggestion
# Describe the layout/UI of the app
```

File: apps/dash-financial-report/app.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    [dcc.Location(id="url", refresh=False), html.Div(id="page-content")]
+)
+
+# Update page
+@app.callback(
+    dash.dependencies.Output("page-content", "children"),
+    [dash.dependencies.Input("url", "pathname")],
+)
+def display_page(pathname):
+    if (
```
Comment (shammamah) [inline apps/dash-financial-report/app.py]: I think you can remove this `if` statement . You've handled all cases in which a pathname that needs to go to a specific view/page is given, and if one of those pages isn't requested, you want to just redirect to the main app. So, get rid of this `if`, change 

```python
    elif pathname == "/dash-financial-report/price-performance":
```

to 

```python
    if pathname == "/dash-financial-report/price-performance":
```

and leave everything else as it is.

File: apps/dash-financial-report/app.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            distributions.create_layout(app),
+            newsReviews.create_layout(app),
+        )
+    else:
+        return overview.create_layout(app)
+
+
+# detail the way that external_css and external_js work and link to alternative method locally hosted
+external_css = [
+    "https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css",
```
Comment (shammamah) [inline apps/dash-financial-report/app.py]: Please download these stylesheets and put them in the `assets` folder!

File: apps/dash-financial-report/app.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+# Describe the layout/ UI of the app
+app.layout = html.Div(
+    [dcc.Location(id="url", refresh=False), html.Div(id="page-content")]
+)
+
+# Update page
+@app.callback(
+    dash.dependencies.Output("page-content", "children"),
+    [dash.dependencies.Input("url", "pathname")],
```
Comment (chriskfwoo) [inline apps/dash-financial-report/app.py]: Just use `Output()` and `Input()`.

File: apps/dash-financial-report/app.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+from dash.dependencies import Input, Output
+import plotly.graph_objs as go
+from utils import Header, make_dash_table
+import pandas as pd
```
Comment (chriskfwoo) [inline apps/dash-financial-report/app.py]: Imports not being used:

```
import plotly.graph_objs as go
from utils import Header, make_dash_table
import pandas as pd
```

File: apps/dash-financial-report/app.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    [dcc.Location(id="url", refresh=False), html.Div(id="page-content")]
+)
+
+# Update page
+@app.callback(
+    dash.dependencies.Output("page-content", "children"),
+    [dash.dependencies.Input("url", "pathname")],
+)
+def display_page(pathname):
+    if pathname == "/dash-financial-report/price-performance":
```
Comment (chriskfwoo) [inline apps/dash-financial-report/app.py]: This is function is totally fine. This is just another alternative solution that I prefer but you don't have to change it 😄 

```Python
    try: 
        lookup_page = {
            "/": overview.create_layout(app),
            "/dash-financial-report": overview.create_layout(app),
            "/dash-financial-report/price-performance": pricePerformance.create_layout(app),
            "/dash-financial-report/portfolio-management": portfolioManagement.create_layout(app),
            "/dash-financial-report/fees": feesMins.create_layout(app),
            "/dash-financial-report/distributions":  distributions.create_layout(app),
            "/dash-financial-report/news-and-reviews": newsReviews.create_layout(app),
            "/dash-financial-report/full-view": (
                overview.create_layout(app),
                pricePerformance.create_layout(app),
                portfolioManagement.create_layout(app),
                feesMins.create_layout(app),
                distributions.create_layout(app),
                newsReviews.create_layout(app),
            ),
        }
        return lookup_page[pathname]
    except KeyError as error:
        # behaves like a 404 page
        return overview.create_layout(app)
```

File: apps/dash-financial-report/app.py [hunk 8]
```diff
@@ -65,9 +48,7 @@ def display_page(pathname):
         return overview.create_layout(app)
 
 
-# detail the way that external_css and external_js work and link to alternative method locally hosted
 external_css = [
-    "https://cdnjs.cloudflare.com/ajax/libs/normalize/7.0.0/normalize.min.css",
     "https://cdnjs.cloudflare.com/ajax/libs/skeleton/2.0.4/skeleton.min.css",
```
Comment (shammamah) [inline apps/dash-financial-report/app.py]: Please get rid of these external stylesheets if you've added them into the `assets` folder!

**Comment by shammamah** [comment]
@celinehuang Is the "full" button supposed to get smaller upon hover? It looks kind of weird. It would be nice if it had the same font/size/etc., but the colours were inverted (so instead of white on red, it would become red on white). 

For a smooth transition, you can use the CSS property `transition-duration`.

File: apps/dash-financial-report/app.py (hunk 8 above)
Comment (celinehuang) [reply apps/dash-financial-report/app.py]: Oops!

**Comment by celinehuang** [comment]
> @celinehuang Is the "full" button supposed to get smaller upon hover? It looks kind of weird. It would be nice if it had the same font/size/etc., but the colours were inverted (so instead of white on red, it would become red on white).
> 
> For a smooth transition, you can use the CSS property `transition-duration`.

I decided to changed the opacity when the mouse is hovering over the link, let me know what you think! Otherwise, I can change it to the colours being inverted

**Comment by shammamah** [comment]
@celinehuang Looks good! Please also use `transition-duration`. You just need to apply it to the selector for the "full view" link/button.

**Pull request merged**